src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -246,6 +246,17 @@
   <data name="AvoidUnsealedAttributesMessage" xml:space="preserve">
     <value>Avoid unsealed attributes</value>
   </data>
+  <data name="AvoidConstArraysTitle" xml:space="preserve">
+    <value>Avoid constant arrays as arguments</value>
+  </data>
+  <data name="AvoidConstArraysDescription" xml:space="preserve">
+    <value>Having a constant array passed as an argument is not ideally performant. Extract constant arrays as 'static readonly'.</value>
+    <comment>{Locked="static readonly"}</comment>
+  </data>
+  <data name="AvoidConstArraysMessage" xml:space="preserve">
+    <value>Make any constant arrays passed as arguments into 'static readonly' fields</value>
+    <comment>{Locked="static readonly"}</comment>
+  </data>
```

Comment:
Thanks for working on this.

I've not reviewed the PR in depth, just enough to see that I think something important to cover hasn't been covered yet.  I'll be curious to see what gets flagged in the dotnet/runtime repo as a result of running this.

If someone has this code:
```C#
ReadOnlySpan<byte> span = new byte[] { 1, 2, 3, 4, 5 };
```
the C# compiler will optimize that to not actually creating an array, but instead storing the data in the binary and just creating a span over that data's address/length.  In such situations, recommending it be extracted to a static readonly byte[] would actually be a deoptimization.  So at least for certain types of arrays, the analyzer would need to to target type and check whether it's being assigned to a `ReadOnlySpan<T>`.

Today those `T`s can be `byte`, `sbyte`, and `bool`, but I'm hopeful that for .NET 7 it'll accommodate any `T` primitive type.  So, it might make sense to avoid firing this suggestion if the target type is `ReadOnlySpan<T>` and `T` is a primitive.

Reply:
I added a check for this. When there is a parameter taking in a `ReadOnlySpan` argument, it ignores the argument.